server/sonar-server/src/test/java/org/sonar/server/es/RecoveryIndexerTest.java

```diff
@@ -187,100 +171,108 @@ public void do_nothing_if_queue_is_empty() {
   }
 
   @Test
-  public void log_exception_on_recovery_failure() {
-    createUnindexedUser();
-    FailingOnceUserIndexer failingOnceUserIndexer = new FailingOnceUserIndexer();
+  public void hard_failures_are_logged_and_do_not_stop_recovery_scheduling() throws Exception {
+    insertItem(FOO_TYPE, "f1");
+
+    HardFailingFakeIndexer indexer = new HardFailingFakeIndexer(FOO_TYPE);
     advanceInTime();
 
-    underTest = newRecoveryIndexer(failingOnceUserIndexer, mockedRuleIndexer);
-    underTest.recover();
+    underTest = newRecoveryIndexer(indexer);
+    underTest.start();
+
+    // all runs fail, but they are still scheduled
+    // -> waiting for 2 runs
+    while (indexer.called.size() < 2) {
+      Thread.sleep(1L);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2925)